LUA/TFS_10/creaturescript playerdeath/playerdeath.lua

```diff
@@ -64,31 +78,22 @@ function onDeath(cid, corpse, killer, mostDamage, unjustified, mostDamage_unjust
 	end
 
 	if byPlayer == 1 then
-		local playerGuild = player:getGuild()
-		if playerGuild then
+		local targetGuild = player:getGuild()
+		targetGuild = targetGuild and targetGuild:getId() or 0
+		if targetGuild ~= 0 then
 			local killerGuild = killerCreature:getGuild()
-			if playerGuild ~= killerGuild and isInWar(cid, killerCreature) then
+			killerGuild = killerGuild and killerGuild:getId() or 0
+			if killerGuild ~= 0 and targetGuild ~= killerGuild and isInWar(cid, killerCreature) then
 				local warId = false
-				resultId = db.storeQuery("SELECT `id` FROM `guild_wars` WHERE `status` = 1 AND ((`guild1` = " .. killerGuild:getId() .. " AND `guild2` = " .. playerGuild:getId() .. ") OR (`guild1` = " .. playerGuild:getId() .. " AND `guild2` = " .. killerGuild:getId() .. "))")
+				resultId = db.storeQuery("SELECT `id` FROM `guild_wars` WHERE `status` = 1 AND ((`guild1` = " .. killerGuild .. " AND `guild2` = " .. targetGuild .. ") OR (`guild1` = " .. targetGuild .. " AND `guild2` = " .. killerGuild .. "))")
 				if resultId ~= false then
 					warId = result.getDataInt(resultId, "id")
 					result.free(resultId)
 				end
 
 				if warId ~= false then
-					db.query("INSERT INTO `guildwar_kills` (`killer`, `target`, `killerguild`, `targetguild`, `time`, `warid`) VALUES (" .. db.escapeString(killerName) .. ", " .. db.escapeString(player:getName()) .. ", " .. killerGuild:getId() .. ", " .. playerGuild:getId() .. ", " .. os.time() .. ", " .. warId .. ")")
-					local wars = db.storeQuery("SELECT `guild_wars`.`id`, (SELECT `limit` FROM `znote_guild_wars` WHERE `znote_guild_wars`.`id` = `guild_wars`.`id`) AS `limit`, (SELECT COUNT(1) FROM `guildwar_kills` WHERE `guildwar_kills`.`warid` = `guild_wars`.`id` AND `guildwar_kills`.`killerguild` = `guild_wars`.`guild1`) guild1_kills, (SELECT COUNT(1) FROM `guildwar_kills` WHERE `guildwar_kills`.`warid` = `guild_wars`.`id` AND `guildwar_kills`.`killerguild` = `guild_wars`.`guild2`) guild2_kills FROM `guild_wars` WHERE (`guild1` = " .. killerGuild:getId() .. " OR `guild2` = " .. killerGuild:getId() .. ") AND `id` = " .. warId .. " AND `status` = 1")
-					if wars ~= false then
-						guildKills1 = result.getDataInt(wars, "guild1_kills")
-						guildKills2 = result.getDataInt(wars, "guild2_kills")
-						fragLimit = result.getDataInt(wars, "limit")
-						result.free(wars)
-					end
-					
-					if guildKills1 >= fragLimit or guildKills2 >= fragLimit then
-						Game.broadcastMessage(string.format("%s has just won the war against %s.", killerGuild:getName(), playerGuild:getName()), MESSAGE_EVENT_ADVANCE)
-						db.query("UPDATE `guild_wars` SET `status` = 4, `ended` = " .. os.time() .. " WHERE `status` = 1 AND `id` = " .. warId)
-					end
```

Comment:
Possible to send channel message upon killing enemy / guild member dies.

Something like this?
````
	local players = Game.getPlayers()
					for _, playerName in ipairs(players) do
						if playerName:getGuild():getId() == targetGuild then
							playerName:sendChannelMessage("", "" .. player:getName() .." was killed by " .. killerName .. ". The new score is ".. guildKills1 ..":".. guildKills2 .." frags (limit ".. fragLimit ..")", TALKTYPE_CHANNEL_R1, CHANNEL_GUILD)
						end
					end
````